src/callcc/boxAssignables.ts

```diff
@@ -13,7 +27,23 @@ function boxVars(path: NodePath<t.Node>, vars: string[]) {
         path.node.init = t.arrayExpression([path.node.init]);
       }
     },
+    AssignmentExpression(path: NodePath<t.AssignmentExpression>) {
+      if ((<any>path.node).__boxVarsInit__) {
+        path.skip();
+      }
+    },
     Identifier(path: NodePath<t.Identifier>) {
+      const parent = path.parent;
+      if (parent.type === "FunctionExpression" ||
+          parent.type === "FunctionDeclaration") {
+        path.skip();
+        return;
+      }
+      if (parent.type === "MemberExpression" &&
```

Comment:
I believe you can avoid an `any` cast here by doing
`if (t.isMemberExpression(parent) && parent.property === path.node) { ...`

`t.isMemberExpression(node)` has return type `node is MemberExpression`, typescript seems to be able to refine on this.